cirq/testing/circuit_compare.py

```diff
@@ -150,17 +116,64 @@ def assert_circuits_with_terminal_measurements_are_equivalent(
         reference: A circuit with the correct function.
         atol: Absolute error tolerance.
     """
-    m1, m2 = _canonicalize_up_to_terminal_measurement_phase(actual, reference)
-
-    assert linalg.allclose_up_to_global_phase(m1, m2, atol=atol), (
-        "Circuit's effect differs from the reference circuit.\n"
-        '\n'
-        'Diagram of actual circuit:\n'
-        '{}\n'
-        '\n'
-        'Diagram of reference circuit with desired function:\n'
-        '{}\n'.format(actual, reference)
-    )
+    measured_qubits_actual = {qubit
+                              for op in actual.all_operations()
+                              if ops.MeasurementGate.is_measurement(op)
+                              for qubit in op.qubits}
+    measured_qubits_reference = {qubit
+                                 for op in reference.all_operations()
+                                 if ops.MeasurementGate.is_measurement(op)
+                                 for qubit in op.qubits}
+    assert actual.are_all_measurements_terminal()
+    assert reference.are_all_measurements_terminal()
+    assert measured_qubits_actual == measured_qubits_reference
+
+    qubits = actual.all_qubits().union(reference.all_qubits())
```

Comment:
It seems we allow the sets of qubits to differ, see e.g. third assert of the first unit test ([l.42 of circuit_compare_test.py](https://github.com/quantumlib/Cirq/blob/master/cirq/testing/circuit_compare_test.py#L42)).

I considered adding an assert that one set is a (possibly improper) subset of the other. There aren't any unit tests that such an assert would fail, but it might be surprising for the users that unused qubits are allowed in one, but not two circuits, so I decided against it in the end. Not a strong opinion.